### PR TITLE
[ttx_diff] Initial refactor for machine-readable output

### PR DESCRIPTION
**Pull request by cmyr**
This splits the generating of the different objects that we compare (ttx tables and layout_normalizer output) from the printing of them, which is the first step towards not printing them.

There is one little addition: it now prints a % representing the diff ratio when two outputs differ.

----

Thinking about what we'll want for the bulk compiler, I realize maybe it's more significantly different than I had thought at first? Basically I'm wondering if, when we're called from the bulk compiler we shouldn't just return some kind of JSON summary that just lists the tables and their difference ratio, and we shouldn't really bother writing all the individual files to disk, since it's all going to be nuked on the next font anyway?

### CODE REVIEW

File: resources/scripts/ttx_diff.py [hunk 1]
```diff
@@ -314,6 +317,76 @@ def reduce_diff_noise(build_dir, fontc, fontmake):
...
+    for tag in sorted(e.tag for e in font_xml.getroot()):
+        table_str = etree.tostring(tables[tag])
+        path = build_dir / f"{compiler}.{tag}.ttx"
+        path.write_bytes(table_str)
+        comparables[tag] = table_str
+
+    return comparables
+
+
+def diff_ratio(text1: str, text2: str):
```
Comment (rsheeter) [inline resources/scripts/ttx_diff.py:381]: Suggest a return type annotation and a comment to indicate how to interpret the value

File: resources/scripts/ttx_diff.py [hunk 2]
```diff
@@ -314,6 +317,76 @@ def reduce_diff_noise(build_dir, fontc, fontmake):
...
+    m = SequenceMatcher(None, text1, text2)
+    return m.quick_ratio()
```
Comment (rsheeter) [inline resources/scripts/ttx_diff.py:383]: Suggest comment explaining why it's ok that we aren't using the real ratio as quick ratio is commented as producing an upper bound on the real ratio

File: resources/scripts/ttx_diff.py (hunk 1 above)
Comment (rsheeter) [reply resources/scripts/ttx_diff.py:381]: (or just switching to use `ratio`)

**Review by rsheeter — APPROVED** [review]

File: resources/scripts/ttx_diff.py (hunk 2 above)
Comment (cmyr) [reply resources/scripts/ttx_diff.py:383]: ratio is just really slow, but I've switched this to use ratio but operate line-wise instead of char-wise.

**Pull request merged**